File: rust/sublist.rs
#[derive(Debug, PartialEq, Eq)]
pub enum Comparison {
    Equal,
    Sublist,
    Superlist,
    Unequal,
}

fn aux<T: PartialEq>(_first_list: &[T], _second_list: &[T]) -> bool {
    if _second_list.len() == 0 {
        return true;
    }
    for j in 0..=_second_list.len() - _first_list.len() {
        let mut res = true;
        for i in 0.._first_list.len() {
            if _first_list[i] != _second_list[j + i] {
                res = false;
                break;
            }
        }
        if res {
            return true;
        }
    }
    false
}

pub fn sublist<T: PartialEq>(_first_list: &[T], _second_list: &[T]) -> Comparison {
    if _first_list.len() == _second_list.len() {
        if aux(_first_list, _second_list) {
            Comparison::Equal
        } else {
            Comparison::Unequal
        }
    } else if _first_list.len() < _second_list.len() {
        if aux(_first_list, _second_list) {
            Comparison::Sublist
        } else {
            Comparison::Unequal
        }
    } else {
        if aux(_second_list, _first_list) {
            Comparison::Superlist
        } else {
            Comparison::Unequal
        }
    }
}

File: python/sublist.py
"""
This exercise stub and the test suite contain several enumerated constants.

Since Python 2 does not have the enum module, the idiomatic way to write
enumerated constants has traditionally been a NAME assigned to an arbitrary,
but unique value. An integer is traditionally used because it’s memory
efficient.
It is a common practice to export both constants and functions that work with
those constants (ex. the constants in the os, subprocess and re modules).

You can learn more here: https://en.wikipedia.org/wiki/Enumerated_type
"""

# Possible sublist categories.
# Change the values as you see fit.
SUBLIST = 1
SUPERLIST = 2
EQUAL = 3
UNEQUAL = 4

def aux(lst1, lst2):
    len1 = len(lst1)
    len2 = len(lst2)
        
    for i in range(len2-len1+1):
        if lst1 == lst2[i:i+len1]:
            return True
        
    return False
        
def sublist(list_one, list_two):
    if list_one == list_two:
        return EQUAL
    
    if aux(list_one, list_two):
        return SUBLIST
    if aux(list_two, list_one):
        return SUPERLIST
        
    return UNEQUAL
